flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/KubernetesScalingRealizer.java

```diff
@@ -63,15 +62,16 @@ public void realizeConfigOverrides(
         }
         FlinkDeployment flinkDeployment = ((FlinkDeployment) context.getResource());
         // Apply config overrides
-        Map<String, String> flinkConf = flinkDeployment.getSpec().getFlinkConfiguration();
-        for (String keyToRemove : configChanges.getRemovals()) {
-            flinkConf.remove(keyToRemove);
-        }
-        flinkConf.putAll(configChanges.getOverrides());
+
+        flinkDeployment.getSpec().getFlinkConfiguration().remove(configChanges.getRemovals());
```

Comment:
Should this be removeAll? Seems like we are removing a single key here which will do nothing

Reply:
actually this uses this method from jackson
```java
public ObjectNode remove(Collection<String> propertyNames) {
        this._children.keySet().removeAll(propertyNames);
        return this;
    }
```
would rather rename our `removeAll` to `remove` so we follow this convention/style